src/com/oltpbenchmark/ThreadBench.java

```diff
@@ -268,13 +268,19 @@ public void run() {
                     return;
                 // Compute the last throughput
                 long measuredRequests = 0;
+                long measuredLatencies = 0;
                 synchronized (testState) {
                     for (Worker w : workers) {
-                        measuredRequests += w.getAndResetIntervalRequests();
+                        measuredLatencies += w.getIntervalLatenciesAverage();
+                        measuredRequests += w.getIntervalRequests();
+                        w.resetIntervalLatencies();
+                        w.resetIntervalRequests();
                     }
                 }
                 double tps = (double) measuredRequests / (double) this.intervalMonitor;
+                double latencyAve = ((double) measuredLatencies / (double) workers.size()) / 1000.0;
```

Comment:
The average of the sum of average latencies is not the same as the average of the latencies. I assume you want the latter, which you could get by just accumulating the latencies in measuredLatencies rather than the average latencies.